helix-core/src/comment.rs

```diff
@@ -94,6 +96,212 @@ pub fn toggle_line_comments(doc: &Rope, selection: &Selection, token: Option<&st
     Transaction::change(doc, changes.into_iter())
 }
 
+fn find_last_non_whitespace_char(text: RopeSlice) -> Option<usize> {
+    text.chars_at(text.len_chars())
+        .reversed()
+        .position(|ch| !ch.is_whitespace())
+        .map(|pos| text.len_chars() - pos - 1)
+}
+
+#[derive(Debug, PartialEq)]
+struct CommentChange {
+    range: Range,
+    open_pos: usize,
+    close_pos: usize,
+    open_margin: bool,
+    close_margin: bool,
+}
+
+fn find_block_comments(
+    open: &str,
+    close: &str,
+    text: RopeSlice,
+    selection: &Selection,
+) -> (bool, Vec<CommentChange>) {
+    let mut commented = true;
+    let mut comment_changes = Vec::with_capacity(selection.len());
+    for range in selection {
+        let selection_slice = range.slice(text);
+        if let (Some(open_pos), Some(close_pos)) = (
+            find_first_non_whitespace_char(selection_slice),
+            find_last_non_whitespace_char(selection_slice),
+        ) {
+            let len = selection_slice.len_chars();
+            let open_len = open.chars().count();
+            let close_len = close.chars().count();
+            let after_open = open_pos + open_len;
+            let before_close = close_pos.saturating_sub(close_len);
+
+            let line_commented = if len >= after_open {
```

Comment:
```suggestion
            let line_commented = if len >= open_Len + close_len {
```

I think that should ge the right length check?

Reply:
Hmm yeah the current check works for all possibilities but the technically correct check I think would be: ``close_pos - open_pos >= open_len + close_len``

Reply:
Ah I thaught that `len = close_pos -open_pos` but it I counts the trimmed white spaces still. Then that check makes more sense but I would prefer the variant I posted (and change the definition of lrn). It's more obvious what it does I think: Check if the slice is large enough to contain a block comment. The current condition could also panic for non-symmetric block comments where the open brace is shorter. HTML jas non-symetric block comments (altough there the open brave is longer) so it's not completely infeasiabke that we would auoprt such comments in the future (or that some user configures the editor that way)

Reply:
Change made👍